src/temperature/weather_service.py

```
@@ -0,0 +1,63 @@
+import httpx
+from fastapi import HTTPException
+
+
+async def fetch_temperature_for_city(city_name: str) -> float:
+
+    geo_url = f"https://geocoding-api.open-meteo.com/v1/search?"
+    geo_params = {"name": city_name}
+
+    async with httpx.AsyncClient() as client:
```

Wrap HTTP calls in a `try/except httpx.RequestError` to handle network/timeouts and raise an appropriate `HTTPException`. Right now network errors will raise unhandled exceptions instead of returning a 5xx HTTPException.